shell/edit/provisioning.cattle.io.cluster/Basics.vue

```diff
@@ -0,0 +1,738 @@
+<script>
+import difference from 'lodash/difference';
+import { mapGetters } from 'vuex';
+import CreateEditView from '@shell/mixins/create-edit-view';
+import FormValidation from '@shell/mixins/form-validation';
+
+import { set, get } from '@shell/utils/object';
+import { Banner } from '@components/Banner';
+import { Checkbox } from '@components/Form/Checkbox';
+import LabeledSelect from '@shell/components/form/LabeledSelect';
+import YamlEditor from '@shell/components/YamlEditor';
+import { LEGACY } from '@shell/store/features';
+import semver from 'semver';
+import { KIND, ELEMENTAL_CLUSTER_PROVIDER } from '@shell/config/elemental-types';
+
+const HARVESTER = 'harvester';
+
+export default {
+  components: {
+    Banner,
+    Checkbox,
+    LabeledSelect,
+    YamlEditor,
+  },
+
+  mixins: [CreateEditView, FormValidation],
+
+  props: {
+    mode: {
+      type:     String,
+      required: true,
+    },
+
+    value: {
+      type:     Object,
+      required: true,
+    },
+
+    provider: {
+      type:     String,
+      required: true,
+    },
+
+    psps: {
+      type:     Object,
+      default:  null,
+      required: false
+    },
+
+    credential: {
+      type:     Object,
+      default:  null,
+      required: false
+    },
+
+    userChartValues: {
+      type:     Object,
+      required: true
+    },
+    cisOverride: {
+      type:     Boolean,
+      required: true
+    },
+
+    cisPsaChangeBanner: {
+      type:     Boolean,
+      required: true
+    },
+    allPSPs: {
+      type:     Array,
+      required: false,
+      default:  null
+    },
+    allPSAs: {
+      type:     Array,
+      required: true
+    },
+
+    addonVersions: {
+      type:     Array,
+      required: false,
+      default:  null
+    },
+    needsPSP: {
+      type:     Boolean,
+      required: true
+    },
+    selectedVersion: {
+      type:     Object,
+      required: true
+    },
+    versionOptions: {
+      type:     Array,
+      required: true
+    },
+    isHarvesterDriver: {
+      type:     Boolean,
+      required: true
+    },
+    isHarvesterIncompatible: {
+      type:     Boolean,
+      required: true
+    },
+    showDeprecatedPatchVersions: {
+      type:     Boolean,
+      required: true
+    }
+
+  },
+
+  data() {
+    return {
+      clusterIsAlreadyCreated: !!this.value.id,
```

Comment:
It looks like this is already defined in rke2.vue - could it be passed into Basics.vue as a prop?